server.py

```diff
@@ -18,8 +19,10 @@ class Server:
     def __init__(self, imagedata):
         self.board = CanvasBoard(imagedata)
         self.lock = threading.Lock()
-    def updateBoard(self,imagedata):
-        self.board = CanvasBoard(imagedata)
+    def updateBoard(self, diff_as_dict):

```

Comment:
you can call it diff and set a type hint here: https://docs.python.org/3/library/typing.html

Reply:
Set type hint as dict & renamed parameter